### DISCUSSION TITLE
Multi-Column Unique Constraint in Directus App

### DISCUSSION THREAD

**Discussion by ravi-2912** [directus/directus#13673] [Feature Requests]
It would be nice to have a **multiple column unique constraint** for SQL tables that can be managed from within the Directus platform. At the minute, if using PostgreSQL, this has to be done by directly writing SQL query and executing it in the database. The query is 

```sql
ALTER TABLE table_name ADD CONSTRAINT uq_c1_c2_c3 UNIQUE ("column_name_1", "column_name_2", "column_name_3")
```

Having a UI for this would be a very usefull and this can be done in the Model page within the Settings Module similar to Duplication Field. 

A UI can be built where the user can check multiple columns to be set as **unique together**.  For an existing **unique together** constraint if the user unchecks a column then the constraint needs to be altered. Likewise, if user unchecks all checked columns then remove the **unique together** constraint. **Unique together** constraint does not need to exist with columns of related tables.

**Comment by piotr-cz** [comment]
While the unique key has to be added manually, Directus shows a validation error when trying to save a record that fails:

> The following fields have invalid values:
> - *constraint_name*: Value has to be unique

I wonder if there is a way to translate the constraint name?

**Comment by rijkvanzanten** [comment]
Also see https://github.com/directus/directus/discussions/14158

**Comment by rijkvanzanten** [comment]
Heya! Thanks for opening this feature request!

This feature request has received over **15 votes** from the community. This means we'll move this feature request to the Under Review state! 

The Core team will schedule a meeting to review this request as soon as possible. The discussion will then be approved or denied. You may or may not be invited to join this meeting with the core team.

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Comment by vszakd** [comment]
Hello,
any news on this?
Thanks.

**Comment by geekyayush** [comment]
Any news on this?
